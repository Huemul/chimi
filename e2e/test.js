const fs = require('fs')
const path = require('path')

// eslint-disable-next-line import/no-extraneous-dependencies
const shell = require('shelljs')

const chimiBin = path.resolve(__dirname, '..', 'bin', 'bin.js')
const fixturesRoot = path.join(__dirname, 'fixtures')

const fixturesDirs = fs
  .readdirSync(fixturesRoot)
  .map(x => path.join(fixturesRoot, x))
  .filter(x => fs.lstatSync(x).isDirectory())

describe('e2e tests', () => {
  // eslint-disable-next-line no-restricted-syntax,prefer-const
  for (let fixtureDir of fixturesDirs) {
    const relativeFixtureDir = path.relative(fixturesRoot, fixtureDir)

    it(`should run ${relativeFixtureDir} as expected`, () => {
      shell.cd(fixtureDir)

      // eslint-disable-next-line global-require,import/no-dynamic-require
      const expectations = require(path.join(fixtureDir, 'expectations.json'))
      const command = `${chimiBin} ${expectations.arguments || ''}`

      const { code, stdout } = shell.exec(command, { silent: true })

      expect(code).toEqual(expectations.status)
      expect(stdout).toMatchSnapshot()
    })
  }
})
