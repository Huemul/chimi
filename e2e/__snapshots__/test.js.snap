// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`e2e tests should run 01-simple-readme as expected 1`] = `
"
Output

----- README.md ----- 

--- Snippet #1 ---
42


 PASS  0-README.md

  Files: 1
  Snippets succeded: 1
"
`;

exports[`e2e tests should run 02-use-globals as expected 1`] = `
"
Output

----- README.md ----- 

--- Snippet #1 ---
42


 PASS  0-README.md

  Files: 1
  Snippets succeded: 1
"
`;

exports[`e2e tests should run 03-multiple-files as expected 1`] = `
"
Output

----- DOCS.md ----- 

--- Snippet #1 ---
42


----- README.md ----- 

--- Snippet #1 ---
42


 PASS  0-DOCS.md
 PASS  0-README.md

  Files: 2
  Snippets succeded: 2
"
`;

exports[`e2e tests should run 04-multiple-snippets as expected 1`] = `
"
Output

----- README.md ----- 

--- Snippet #1 ---
42


--- Snippet #2 ---
42


 PASS  0-README.md
 PASS  1-README.md

  Files: 1
  Snippets succeded: 2
"
`;

exports[`e2e tests should run 05-simple-fail as expected 1`] = `
"
Output

----- README.md ----- 


Errors

----- README.md ----- 

--- Snippet #1 ---

/snippet.js:1
console.log(answer)
^
ReferenceError: answer is not defined
    at Object.<anonymous> (/snippet.js:1:1)
    at Module._compile (module.js:624:30)
    at Object.Module._extensions..js (module.js:635:10)
    at Module.load (module.js:545:32)
    at tryModuleLoad (module.js:508:12)
    at Function.Module._load (module.js:500:3)
    at Function.Module.runMain (module.js:665:10)
    at startup (bootstrap_node.js:187:16)
    at bootstrap_node.js:608:3


 FAIL  0-README.md

  Files: 1
  Snippets rejected: 1
"
`;
